### ISSUE TITLE
Request for free bagslots/total bagslots on item tooltip

### ISSUE THREAD

**Issue by Harze2k** [Xruptor/BagSync#135]
Hi!

Glad to see you have updated this awesome addon! 
A small (perhaps?) request. 
Would it be possible to when you mouse over an item you see who has it AND also how many free sots that char has. 
Could also customize so you can just see empty slots or empty/total or any other clever combination :)
I have 20 chars on 2 accounts and knowing how many free slots that character has on a tooltip would save soo much time! 
+1 Internets to you if you can fix this :D

**Comment by Xruptor** [comment]
I will admit this is somewhat of an odd request.  I never figured that knowing how much empty space a character had would be valuable information to display. Hmm... Could you provide me some scenarios where this may be beneficial?  That way I can see it from your viewpoint and a different angle.  I'm currently working on another project, but I will take a look at your request when I get a spare moment.  I cannot promise you that I will implement, but I do assure you I will review and look into it.

**Comment by Harze2k** [comment]
Cheers for the quick answer! Yes maybe I can explain it better. I am a horder really and have a lot of characters, some have 0 bag spaces left and some perhaps 20-50 slots free. It's always a struggle to find where to ship items in quantity around since so many times that guy just had no space left in bank and bags. 

I think at least some recognize this issue :) if you can get around to it at some point it would be awesome! And if you can throw in a check for bank slots also would be the cherry on top! 

Thanx! /Martin

**Comment by Harze2k** [comment]
I know there are adons that you can see this information but it's hidden by tabs and layers of clicking and trying to remember where it was. A quick glance on the tool tip would solve this really nice! 😊

**Comment by Xruptor** [comment]
Okay I think I understand now.  I only really see one major issue with this request though.  Lets say you have two characters named Jane and the other Bob.  When you login as the Jane and hover an item, you notice that Bob has 2 slots left.  So you send Bob 2 items in the mail. Once you send the mail you hover over the exact item again in Jane's inventory, only for it to show that Bob still has 2 slots left.

So why did this occur?  Well because you would have to logoff as Jane and then login as Bob for that character data to be properly updated in the database. 

Inventory and character data is scanned and saved for BagSync when the character is logged in.  Its impossible to scan a character while they are offline.  The only way I would have to get around this, would be to scan the mailbox for items sent to Bob while logged in as Jane.  Then calculate the amount of items sent and substract that from Bobs bag space count in the database.  That would add a layer of complexity that I don't really want to implement right now.

So in a nutshell, yes I can grab the empty bag spaces of the current character and store it in the DB.  However, when you login on another character it will always display that amount of empty space, until you return to that character.  Do you see the issue?
If I can think of a more smooth way around this, I may implement.  Otherwise, having another extra layer of scanning for the mailbox just for sent items isn't really something I want to do.

**Comment by Xruptor** [comment]
After reviewing this again, I don't feel this is something I'd like to implement at the moment.  Hopefully I explained why as it would just add more complexity right now to the addon I just don't want to add at the moment.  I also don't have a subscription LOL.
